resources/type-system/inference.md

```diff
@@ -1139,181 +1204,152 @@ as follows:
-both instances of the type `bool`._
+### Booleans
 
-## Strings
+Expression inference of a boolean literal `e` (`true` or `false`), regardless of
+context, produces the elaborated expression `e`, with static type `bool`.
 
-Expression type inference of a string literal `s` produces an expression
-artifact `m` with static type `String`, where `m` is determined as follows:
+_The runtime behavior of a boolean literal is to evaluate to an instance of the
+type `bool`, so soundness is satisfied._
 
-- For each _stringInterpolation_ `s_i` inside `s`, in source order:
+### Strings
 
-  - Define `m_i` as follows:
+Expression inference of a string literal `s`, regardless of context, produces an
+elaborated expression `m` with static type `String`, where `m` is determined as
+follows:
 
-    - If `s_i` takes the form '`${`' `e` '`}`':
+- If `s` contains no _stringInterpolations_, then let `m` be `s`. _The runtime
+  behavior of a string liveral with no _stringInterpolations_ is to evaluate to
+  an instance of the type `String`, so soundness is satisfied._
 
-      - Let `m_i` be the result of performing expression type inference on `e`,
-        in context `_`.
+- Otherwise:
 
-    - Otherwise, `s_i` takes the form '`$e`', where `e` is either `this` or an
-      identifier that doesn't begin with `$`, so:
+  - For each _stringInterpolation_ `s_i` inside `s`, in source order:
 
-      - Let `m_i` be the result of performing expression type inference on `e`,
-        in context `_`.
+    - Define `m_i` as follows:
 
-  - Let `T_i` be the static type of `m_i`.
+      - If `s_i` takes the form '`${`' `e` '`}`':
 
-- Let `m` be an expression artifact whose runtime behavior is as follows:
+        - Let `m_i` be the result of performing expression inference on `e`, in
+          context `_`.
 
-  - For each `i`, in order:
+      - Otherwise, `s_i` takes the form '`$e`', where `e` is either `this` or an
+        identifier that doesn't begin with `$`, so:
 
-    - Execute expression artifact `m_i`, and let `o_i` be the resulting value.
+        - Let `m_i` be the result of performing expression inference on `e`, in
+          context `_`.
 
-    - Invoke the `toString` method on `o_i`, with no arguments, and let `r_i` be
-      the return value. _Note that since both `Object.toString` and
+    - Let `n_i` be `m_i.toString()`. _Since both `Object.toString` and
       `Null.toString` are declared with a return type of `String`, it follows
-      from return value soundness that `r_i` will have a runtime type of
-      `String`)._
+      that the static type of `n_i` is `String`._
 
-  - Let `r` be an instance of `String` whose characters are the result of
-    concatenating together the `r_i` strings, interspersing with any
-    non-interpolation characters in `s`.
+  - Let `m` be `@CONCAT(parts)`, where `parts` is composed of simple string
+    literals representing the portions of `s` that are __not__
```

Comment:
I’m happy to clarify whether these simple string literals were meant to be elaborated or unelaborated expressions, but I’m not sure what information I would need to add. Can you say more about why it unclear to you? I meant for them to be elaborated expressions, and I thought it was clear because (a) they are part of the output of type inference (and type inference produces elaborated expressions as its output), (b) at the top of this section, I say "produces an elaborated expression `m`", and (c) in the definition of `@CONCAT`, I said that each of its arguments is an elaborated expression.

As for whether they have a static type, as I mentioned in a previous comment, my mental model is that elaborated expressions inherit the static typing defined in `dartLangSpec.tex`, so I think that these simple string literals have a static type of `String`.